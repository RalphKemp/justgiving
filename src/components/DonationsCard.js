import React from 'react';
import styled from 'styled-components';
import { media } from '../helpers/sizing';
import uuidv1 from 'uuid/v1';
import dateFormat from "dateformat";

const DonationsCardContainer = styled.div`
  position: relative;
  background-color: rgb(0, 119, 45);
  width: 100%;
  margin-top: 10px;
  height: auto;
  border-radius: 5px;
  font-family: 'Roboto', sans-serif;
  font-weight: 400;
  color: white;
  display: flex;
  flex-direction: column !important;
  padding-bottom: 15px;
  ${media.tablet`
      width: 40%;
      margin-left: 17px;
  `};
`;

const DonationsHeader = styled.div`
  font-weight: 500;
  text-align: center;
  font-size: 20px;
  position: absolute;
  margin-top: 18px;
  top: -50px;
  left: 75px;
  color: black;
  font-size: 23px;
  ${media.mid`
    left: 0px;
  `};
`;

const AllDonations = styled.div`
  display: flex;
  flex-direction: column !important;
`;

const IndividualDonation = styled.div`
  font-size: 12px;
  background-color: white;
  color: black;
  margin: 5px;
  border-radius: 10px;
  padding: 5px;
  overflow: hidden;
`;

// for some reason date from api is coming back weird, so have to substr()

const DonationsCard = ({ donations }) => {
  return (
    <DonationsCardContainer>
      <DonationsHeader>
        <p>Donations</p>
      </DonationsHeader>
      <AllDonations>
        {donations.donations.map(donation => {
          const time = donation.donationDate.substr(6, 13);
          const donationDate = new Date(parseInt(time)).toString();
          return (
            <IndividualDonation key={uuidv1()}>
              {dateFormat(donationDate, "dddd, mmmm dS, yyyy, h:MM:ss TT")}
              <br />
              {donation.donorDisplayName} donated Â£{donation.amount}
              <br />
              {donation.message}
              <br />
              {donation.imageUrl}
            </IndividualDonation>
          );
        })}
      </AllDonations>
    </DonationsCardContainer>
  );
};

export default DonationsCard;
