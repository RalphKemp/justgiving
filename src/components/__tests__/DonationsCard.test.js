import React from 'react';
import { shallow } from 'enzyme';
// using curly braces to import component without state.
import { DonationsCard } from '../DonationsCard.js';

describe('DonationsCard', () => {

  const defaultProps = {
    donations: [
      {
        amount: 55,
        donationDate: 'march 2018',
        donorDisplayName: 'ralph',
        message: 'Our deepest condolences'
      },
      {
        amount: 55,
        donationDate: 'march 2018',
        donorDisplayName: 'ralph',
        message: 'Our deepest condolences'
      },
      {
        amount: 55,
        donationDate: 'march 2018',
        donorDisplayName: 'ralph',
        message: 'Our deepest condolences'
      },
      {
        amount: 55,
        donationDate: 'march 2018',
        donorDisplayName: 'ralph',
        message: 'Our deepest condolences'
      }
    ]
  }

  it('renders correctly', () => {
    const wrapper = shallow(<DonationsCard donations={defaultProps}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
