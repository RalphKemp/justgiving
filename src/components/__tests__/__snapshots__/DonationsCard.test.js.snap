// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonationsCard renders correctly 1`] = `
<styled.div>
  <styled.div>
    <p>
      Recent Donations
    </p>
  </styled.div>
  <styled.div>
    <styled.div
      key="Thu Jan 01 1970 01:00:02 GMT+0100 (Greenwich Mean Time)"
    >
      <Styled(styled.div)>
        Thursday, January 1st, 1970
      </Styled(styled.div)>
      <Styled(styled.div)>
        <div>
          ralph
           donated 
          <span>
            £
            55
          </span>
        </div>
      </Styled(styled.div)>
      <Styled(styled.div)>
        <p>
           - "
          Our deepest condolences
          "
        </p>
      </Styled(styled.div)>
    </styled.div>
    <styled.div
      key="Thu Jan 01 1970 01:00:02 GMT+0100 (Greenwich Mean Time)"
    >
      <Styled(styled.div)>
        Thursday, January 1st, 1970
      </Styled(styled.div)>
      <Styled(styled.div)>
        <div>
          ralph
           donated 
          <span>
            £
            55
          </span>
        </div>
      </Styled(styled.div)>
      <Styled(styled.div)>
        <p>
           - "
          Our deepest condolences
          "
        </p>
      </Styled(styled.div)>
    </styled.div>
    <styled.div
      key="Thu Jan 01 1970 01:00:02 GMT+0100 (Greenwich Mean Time)"
    >
      <Styled(styled.div)>
        Thursday, January 1st, 1970
      </Styled(styled.div)>
      <Styled(styled.div)>
        <div>
          ralph
           donated 
          <span>
            £
            55
          </span>
        </div>
      </Styled(styled.div)>
      <Styled(styled.div)>
        <p>
           - "
          Our deepest condolences
          "
        </p>
      </Styled(styled.div)>
    </styled.div>
    <styled.div
      key="Thu Jan 01 1970 01:00:02 GMT+0100 (Greenwich Mean Time)"
    >
      <Styled(styled.div)>
        Thursday, January 1st, 1970
      </Styled(styled.div)>
      <Styled(styled.div)>
        <div>
          ralph
           donated 
          <span>
            £
            55
          </span>
        </div>
      </Styled(styled.div)>
      <Styled(styled.div)>
        <p>
           - "
          Our deepest condolences
          "
        </p>
      </Styled(styled.div)>
    </styled.div>
  </styled.div>
</styled.div>
`;
